JavaSource/org/unitime/timetable/gwt/client/departments/DepartmentsPage.java

```diff
@@ -0,0 +1,286 @@
+/*
+ * Licensed to The Apereo Foundation under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ *
+ * The Apereo Foundation licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * 
+ */
+package org.unitime.timetable.gwt.client.departments;
+
+import java.util.logging.Logger;
+
+import org.unitime.timetable.gwt.client.ToolBox;
+import org.unitime.timetable.gwt.client.page.UniTimeNotifications;
+import org.unitime.timetable.gwt.client.page.UniTimePageLabel;
+import org.unitime.timetable.gwt.shared.DepartmentInterface;
+import org.unitime.timetable.gwt.shared.DepartmentInterface.GetDepartmentsRequest;
+import org.unitime.timetable.gwt.shared.DepartmentInterface.DepartmentsDataResponse;
+import org.unitime.timetable.gwt.client.widgets.LoadingWidget;
+import org.unitime.timetable.gwt.client.widgets.SimpleForm;
+import org.unitime.timetable.gwt.client.widgets.UniTimeHeaderPanel;
+import org.unitime.timetable.gwt.client.widgets.UniTimeWidget;
+import org.unitime.timetable.gwt.client.widgets.UniTimeTable.MouseClickListener;
+import org.unitime.timetable.gwt.client.widgets.UniTimeTable.TableEvent;
+import org.unitime.timetable.gwt.command.client.GwtRpcService;
+import org.unitime.timetable.gwt.command.client.GwtRpcServiceAsync;
+import org.unitime.timetable.gwt.resources.GwtMessages;
+
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.event.dom.client.ClickEvent;
+import com.google.gwt.event.dom.client.ClickHandler;
+import com.google.gwt.user.client.rpc.AsyncCallback;
+import com.google.gwt.user.client.ui.CheckBox;
+import com.google.gwt.user.client.ui.Composite;
+import com.google.gwt.user.client.ui.SimplePanel;
+
+public class DepartmentsPage extends Composite {
+	protected static final GwtMessages MESSAGES = GWT.create(GwtMessages.class);
+	protected static GwtRpcServiceAsync RPC = GWT.create(GwtRpcService.class);
+
+	private SimplePanel iPanel;
+	private SimpleForm iListDepartmentsForm;
+	private UniTimeHeaderPanel iListDepartmentsHeader, iListDepartmentsFooter;
+	private DepartmentsTable iDepartmentsTable;
+	private DepartmentsTable iDepartmentsFullTable;
```

Comment:
You do not really need to create an additional table just to remember all the departments. You can just remember the last received DepartmentsDataResponse and iterate DepartmentsDataResponse.getDepartments() to check for uniqueness.